numpy/core/arrayprint.py

```diff
@@ -597,8 +634,8 @@ def _formatArray(a, format_function, rank, max_line_len,
                           summary_insert).rstrip()+']\n'
     return s
 
-class FloatFormat(object):
-    def __init__(self, data, precision, suppress_small, sign=False):
+class FloatingFormat(object):
```

Comment:
Same comment about leaving a public `FloatFormat` object visible, assuming we're considering this public API.

Docstring saying `For np.floating objects` or something